easy-random-core/src/main/java/org/jeasy/random/util/ReflectionUtils.java

```diff
@@ -610,6 +596,27 @@ private static void rejectUnsupportedTypes(Class<?> type) {
         }
     }
 
+    private static Method getSetter(Class<?> beanType, String propertyName, Class<?> propertyType) {
+        String setterName = "set" + propertyName.substring(0, 1).toUpperCase(ENGLISH) + propertyName.substring(1);
+        // implementation note: we go from subclass to super-class order - hence getDeclaredMethods
+        for (Method method : beanType.getDeclaredMethods()) {
```

Comment:
Following KISS principle, this may be simplified to `beanType.getMethods()` instead (in this case the recursive super call can be omitted).

I've validated this change on about 200 model classes in my projects (I had good coverage with cases - so I've found issues described in #437), so at least I'm sure that it's better than it was in `4.2.0` and in `master`

Reply:
Decided to simplify (see second commit) and unify with `getReadMethod`